smolyay/basis.py

```diff
@@ -131,7 +131,7 @@ def __call__(self, x):
             if input is outside the domain [-1,1]
 
         """
-        if x > 1 or x < -1:
+        if numpy.any([numpy.greater(x,1),numpy.less(x,-1)]):
```

Comment:
This should avoid an array conversion and is the analog of what was here before:
```suggestion
        if numpy.any(numpy.greater(x, 1)) or numpy.any(numpy.less(x, -1)):
```
If `x` could be required to be a NumPy array, you could just use `x > 1` and `x < -1` instead the functions, but this is a good compatible work around.

Please apply to other spots below.